crates/stack/src/interface.rs

```diff
@@ -1,89 +1,149 @@
-use std::collections::BTreeMap;
+use managed::ManagedSlice;
+use smoltcp::iface::{Config, Interface, Route, SocketHandle, SocketSet};
+use smoltcp::socket::AnySocket;
+use smoltcp::time::Instant;
+use smoltcp::wire::{EthernetAddress, HardwareAddress, IpAddress, IpCidr};
 use std::io::Write;
 
-use managed::{ManagedMap, ManagedSlice};
-use ya_smoltcp::iface::{Interface, InterfaceBuilder, NeighborCache, Route, Routes};
-use ya_smoltcp::wire::{EthernetAddress, HardwareAddress, IpAddress, IpCidr};
-
 use crate::device::CaptureDevice;
 
-pub type CaptureInterface<'a> = Interface<'a, CaptureDevice>;
+pub struct CaptureInterface<'a> {
```

Comment:
smoltcp no longer has a type aggregating `Interface` with `CaptureDevice` and `SocketSet`.